bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/RawType.java

```diff
@@ -22,32 +22,45 @@
  */
 public class RawType implements PrimitiveType, State {
 
+    private static final String DEFAULT_MIME_TYPE = "image/jpeg";
+
     protected byte[] bytes;
+    protected String mimeType;
 
     public RawType() {
         this(new byte[0]);
     }
 
     public RawType(byte[] bytes) {
+        this(bytes, null);
+    }
+
+    public RawType(byte[] bytes, String mimeType) {
         this.bytes = bytes;
+        this.mimeType = (mimeType == null) ? DEFAULT_MIME_TYPE : mimeType;
     }
 
     public byte[] getBytes() {
         return bytes;
     }
 
+    public String getMimeType() {
+        return mimeType;
+    }
+
     public static RawType valueOf(String value) {
-        return new RawType(PortableBase64.getDecoder().decode(value));
+        String mimeType = value.substring(value.indexOf("data:") + 5, value.indexOf(";base64,"));
+        return new RawType(PortableBase64.getDecoder().decode(value.substring(value.indexOf(",") + 1)), mimeType);
     }
 
     @Override
     public String toString() {
-        return String.format("raw type: %d bytes", bytes.length);
+        return String.format("raw type (%s): %d bytes", mimeType, bytes.length);
     }
 
     @Override
     public String toFullString() {
-        return PortableBase64.getEncoder().encode(bytes);
+        return String.format("data:%s;base64,%s", mimeType, PortableBase64.getEncoder().encode(bytes));

```

Comment:
Same question as above: since this assumes HTML formatting, this is less a `RawType` and more specific to representing raw image data in a way understood in HTML.